core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -577,9 +577,12 @@ private RexNode simplifyIsNull(RexNode a) {
   private RexNode simplifyCoalesce(RexCall call) {
     final Set<String> digests = new HashSet<>();
     final List<RexNode> operands = new ArrayList<>();
+    final RexSimplify simplify = withUnknownAsFalse(false);
```

Comment:
Is this really needed for CALCITE-2580?

Reply:
In UnknownAsFalse mode,for "Coalesce(null,x)",  "null" will simplify to "false" without this code which lead to the wrong result.

Reply:
makes sense